src/nc_conf.c

```diff
@@ -360,6 +384,158 @@ conf_dump(struct conf *cf)
             s = array_get(&cp->server, j);
             log_debug(LOG_VVERB, "    %.*s", s->len, s->data);
         }
+
+        nserver = array_n(&cp->sentinel);
+        if (nserver > 0) {
+            log_debug(LOG_VVERB, "  sentinels: %"PRIu32"", nserver);
+
+            for (j = 0; j < nserver; j++) {
+                s = array_get(&cp->sentinel, j);
+                log_debug(LOG_VVERB, "    %.*s", s->len, s->data);
+            }
+        }
+    }
+}
+
+static rstatus_t
+conf_write(FILE *fh, const char *fmt, ...)
+{
+    size_t size, n;
+    int len;
+    char buf[256];
+    va_list args;
+
+    len = 0;
+    size = sizeof(buf);
+    va_start(args, fmt);
+    len = nc_vscnprintf(buf, size, fmt, args);
+    va_end(args);
+
+    buf[len++] = '\n';
+
+    n = fwrite(buf, (size_t)len, (size_t)1, fh);
+    if (n == 0) {
+        return NC_ERROR;
+    }
+
+    return NC_OK;
+}
+
+void
+conf_rewrite(struct context *ctx)
+{
+    uint32_t i, j, npool, nserver;
+    struct conf *cf;
+    struct conf_pool *cp;
+    struct conf_server *cs;
+    struct string true_str, false_str, bool_str;
+    FILE *fh;
+    char temp_conf_file[256];
+
+    cf = ctx->cf;
+
+    npool = array_n(&cf->pool);
+    if (npool == 0) {
+        return;
+    }
+
+    log_debug(LOG_VVERB, "%"PRIu32" pools in configuration file '%s'", npool,
+              cf->fname);
+
+    /* open temp conf file to rewrite */
+    nc_snprintf(temp_conf_file, 256, "%s.%d.temp", cf->fname, (int)getpid());
+    fh = fopen(temp_conf_file,"w");
+    if (fh == NULL) {
+        log_error("conf: failed to open temp configuration '%s': %s",
+                  temp_conf_file, strerror(errno));
+        return;
+    }
+
+    string_set_text(&true_str, "true");
+    string_set_text(&false_str, "false");
+
+    for (i = 0; i < npool; i++) {
+        cp = array_get(&cf->pool, i);
+
+        conf_write(fh, "%.*s:", cp->name.len, cp->name.data);
```

Comment:
This should check the error code of conf_write, in case the disk fills up or other errors occur, to avoid replacing configs with an empty file?

`#define conf_write_or_fail(...) if (conf_write(__VA_ARGS__) != NC_OK) { goto error; }`

See https://github.com/ifwe/twemproxy/pull/38 for an example of how that could be done